lib/logstash/filters/mutate.rb

```diff
@@ -428,12 +428,13 @@ def merge(event)
           next
         end
 
+        # No need to test the other
         if dest_field_value.is_a?(Hash)
-          # No need to test the other
-          event[dest_field].update(added_field_value)
+          # do not use event[dest_field].update to avoid assuming the in-place mutability of field values
```

Comment:
Maybe rephrase it somehow indicating that field values cannot be mutated in-place, only assigned. I'm not sure how to phrase it, but the current phrasing feels flimsy. We're moving to a model where an event field value 'get' will return a copied (or immutable) value, right?